include/ola/client/ClientTypes.h

```diff
@@ -270,30 +280,39 @@ class OlaUniverse {
   OlaUniverse(unsigned int id,
               merge_mode m,
               const std::string &name,
-              unsigned int input_port_count,
-              unsigned int output_port_count,
+              std::vector<OlaInputPort> input_ports,
```

Comment:
Pass vectors as const references, that avoids the copy. e.g.

const std::vector<OlaInputPort> &input_ports.

You could even pass in pointers and have the object take ownership. That would avoid a second copy:

std::auto_ptr<std::vector<OlaInputPort>> m_input_ports;

OlaUniverse(unsigned int id,
                     const std::vector<OlaInputPort> *input_ports)
    : m_input_ports(input_ports) {
}